### PR TITLE
Replaced torch.distributed.launch with torchrun

### LINKED ISSUE #199: Update torch.distributed.launch to torchrun
`torch.distributed.launch` is be deprecated, let's use recommended launcher. Related ignite issue: https://github.com/pytorch/ignite/issues/2415

### PR DESCRIPTION
**Pull request by sayantan1410**
Fix #199 
- [X] Bug fix

Let me know if I have made some mistake, will correct it out.

### CODE REVIEW

File: src/templates/template-text-classification/README.md [hunk 1]
```diff
@@ -24,26 +24,20 @@ pip install -r requirements.txt --progress-bar off -U
 - Execute on master node
 
 ```sh
-python -m torch.distributed.launch \
+torchrun main.py \
   --nproc_per_node #:::= it.nproc_per_node :::# \
   --nnodes #:::= it.nnodes :::# \
   --node_rank 0 \
-  --master_addr #:::= it.master_addr :::# \
```
Comment (vfdev-5) [inline src/templates/template-text-classification/README.md:31]: @sayantan1410 why do you remove `--master_addr` and `--master_port` args ?

File: src/templates/template-text-classification/README.md [hunk 2]
```diff
@@ -24,26 +24,20 @@ pip install -r requirements.txt --progress-bar off -U
...
-  --master_port #:::= it.master_port :::# \
-  --use_env main.py \
```
Comment (vfdev-5) [inline src/templates/template-text-classification/README.md:33]: We should keep `main.py` as executable script otherwise it does not make any sense

**Review by vfdev-5 — COMMENTED** [review]
@sayantan1410 thanks for the PR, I left comments. Please also try to check proposed updates locally or in colab on cpus.

File: src/templates/template-text-classification/README.md (hunk 1 above)
Comment (sayantan1410) [reply src/templates/template-text-classification/README.md:31]: Actually I was going through this https://pytorch.org/docs/stable/elastic/train_script.html#elastic-train-script, 
and it stated that these two args are not needed to be passes manually.

**Comment by sayantan1410** [comment]
> Please also try to check proposed updates locally or in colab on cpus.

Actually, I was trying to check it by running the changes on colab, but when I am clicking open in colab, it is reverting me back to the home page of the code-generator, specifically in to localhost:3000/create

**Comment by vfdev-5** [comment]
> > Please also try to check proposed updates locally or in colab on cpus.
> 
> Actually, I was trying to check it by running the changes on colab, but when I am clicking open in colab, it is reverting me back to the home page of the code-generator, specifically in to localhost:3000/create

Open in Colab works only on netlify deployed app.

File: src/templates/template-text-classification/README.md (hunk 1 above)
Comment (vfdev-5) [reply src/templates/template-text-classification/README.md:31]: This docs about elastic training and in any case user should provide master address/port for multi-node training. torchrun can't guess it by itself. In this elastic case, we need to provide `rdzv_endpoint`. 
For code-generator we do not need to provide elastic option, just replacing depreacted torch.distributed.launch.

**Comment by sayantan1410** [comment]
@vfdev-5 hey, I made the changes and it passed the tests, let me know if this is good .

File: src/templates/template-common/README.md [hunk 3]
```diff
@@ -3,42 +3,42 @@
...
 #::: if (it.nnodes > 1 && it.master_addr && it.master_port) { :::#
 
-### Multi Node, Multi GPU Training (`torch.distributed.launch`) (recommended)
+### Multi Node, Multi GPU Training (`torch.distributed.run`) (recommended)
 
 - Execute on master node
 
 ```sh
-python -m torch.distributed.launch \
+python -m torch.distributed.run \
```
Comment (vfdev-5) [inline src/templates/template-common/README.md:11]: Let's use torchrun as it is much shorter syntax

File: src/templates/template-common/README.md [hunk 4]
```diff
@@ -3,42 +3,42 @@
...
   --nproc_per_node #:::= it.nproc_per_node :::# \
   --nnodes #:::= it.nnodes :::# \
   --node_rank 0 \
   --master_addr #:::= it.master_addr :::# \
   --master_port #:::= it.master_port :::# \
-  --use_env main.py \
+    main.py \
```
Comment (vfdev-5) [inline src/templates/template-common/README.md:17]:
```suggestion
  main.py \
```

**Comment by sayantan1410** [comment]
@vfdev-5  made the changes as you suggested, please check !!

File: src/templates/template-common/README.md [hunk 5]
```diff
@@ -3,42 +3,42 @@
 #::: if (it.nproc_per_node) { :::#
 #::: if (it.nnodes > 1 && it.master_addr && it.master_port) { :::#
 
-### Multi Node, Multi GPU Training (`torch.distributed.launch`) (recommended)
+### Multi Node, Multi GPU Training (`torch.distributed.run`) (recommended)
```
Comment (vfdev-5) [inline src/templates/template-common/README.md:6]:
```suggestion
### Multi Node, Multi GPU Training (`torchrun`) (recommended)
```

File: src/templates/template-common/README.md [hunk 6]
```diff
@@ -3,42 +3,42 @@
...
   --master_port #:::= it.master_port :::# \
-  --use_env main.py \
+  main.py \
   --backend #:::= it.backend :::#
 ```
 
 #::: } else { :::#
 
-### Multi GPU Training (`torch.distributed.launch`) (recommended)
+### Multi GPU Training (`torch.distributed.run`) (recommended)
```
Comment (vfdev-5) [inline src/templates/template-common/README.md:36]:
```suggestion
### Multi GPU Training (`torchrun`) (recommended)
```

and others

File: src/templates/template-text-classification/README.md [hunk 7]
```diff
@@ -19,42 +19,42 @@ pip install -r requirements.txt --progress-bar off -U
...
 
 - Execute on worker nodes
 
 ```sh
-python -m torch.distributed.launch \
+torchrun \
   --nproc_per_node #:::= it.nproc_per_node :::# \
   --nnodes #:::= it.nnodes :::# \
-  --node_rank <node_rank> \
+  --node_rank 0 \
```
Comment (vfdev-5) [inline src/templates/template-text-classification/README.md:43]: Let's revert this change

File: src/templates/template-vision-segmentation/README.md [hunk 8]
```diff
@@ -19,42 +19,42 @@ pip install -r requirements.txt --progress-bar off -U
...
 
 - Execute on worker nodes
 
 ```sh
-python -m torch.distributed.launch \
+torchrun \
   --nproc_per_node #:::= it.nproc_per_node :::# \
   --nnodes #:::= it.nnodes :::# \
-  --node_rank <node_rank> \
+  --node_rank 0 \
```
Comment (vfdev-5) [inline src/templates/template-vision-segmentation/README.md:43]: revert that as well

**Comment by vfdev-5** [comment]
Thanks for the update @sayantan1410 , now we can also fix distributed tests: 
https://github.com/pytorch-ignite/code-generator/blob/main/scripts/run_tests.sh

**Comment by sayantan1410** [comment]
@vfdev-5 made the changes, please check !!

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @sayantan1410 !
